router/http_test.go

```diff
@@ -45,18 +48,38 @@ BAwwCgYIKwYBBQUHAwEwDwYDVR0TAQH/BAUwAwEB/zA9BgNVHREENjA0ggtleGFt
 cGxlLmNvbYINKi5leGFtcGxlLmNvbYcEfwAAAYcQAAAAAAAAAAAAAAAAAAAAATAL
 BgkqhkiG9w0BAQsDQQBJxy1zotHYLZpyoockAlJWRa88hs1PrroUNMlueRtzNkpx
 9heaebvotwUkFlnNYJZsfPnO23R0lUlzLJ3p1RNz
------END CERTIFICATE-----`)
-
-// localhostKey is the private key for localhostCert.
-var localhostKey = []byte(`-----BEGIN RSA PRIVATE KEY-----
+-----END CERTIFICATE-----`,
+		PrivateKey: `-----BEGIN RSA PRIVATE KEY-----
 MIIBOQIBAAJBALfSVyYOnwjMbwSnywrpB+3ZxdZyHQgbrq418VTOvRNvvext2gwU
 yJu617cYpQGImwcMwbYKj81FZ4G5ituXILkCAwEAAQJAXvmhp3skdkJSFgCv6qou
 O5kqG7uH/nl3DnG2iA/tJw3SlEPftQyzNk5jcIFSxvr8pu1pj+L1vw5pR68/7fre
 xQIhAMM0/bYtVbzW+PPjqAev3TKhMyWkY3t9Qvw5OtgmBQ+PAiEA8RGk9OvMxBbR
 8zJmOXminEE2VVE1VF0K0OiFLDG+JzcCIHurptE0B42L5E0ffeTg1hKtben7K8ug
 oD+LQmyOKcahAiB05Btab2QQyQfwpsWOpP5GShCwefoj+CGgfr7kWRJdLQIgTMZe
 ++SKD8ascROyDnZ0Td8wbrFnO0YRPEkwlhn6h0U=
------END RSA PRIVATE KEY-----`)
+-----END RSA PRIVATE KEY-----`,
+	},
+}
+
+func tlsConfigForDomain(domain string) ([]byte, []byte) {
+	domain = strings.ToLower(domain)
+	if d, _, err := net.SplitHostPort(domain); err == nil {
+		domain = d
+	}
+	if strings.HasSuffix(domain, ".example.com") {
+		domain = "example.com"
+	}
+	if c, ok := tlsCerts[domain]; ok {
+		return []byte(c.CACert), []byte(c.PrivateKey)
+	}
+	fmt.Printf("Generating TLS Cert for %s...\n", domain)
+	c, err := tlscert.Generate([]string{domain})
+	if err != nil {
+		panic(err)
+	}
+	tlsCerts[domain] = c
```

Comment:
Writes and reads to this map should be wrapped in a mutex, tests could run concurrently depending on the configuration.